buildenv/jenkins/JenkinsfileBase

```diff
@@ -858,16 +858,23 @@ def uploadToArtifactory(pattern) {
 def addGrinderLink() {
 	String url = "${HUDSON_URL}job/Grinder/parambuild/?"
 	int i = 1;
+	def labelValue = ""
 	params.each { key, value ->
 		value = URLEncoder.encode(value.toString(), "UTF-8")
 		url += "${key}=${value}"
 		if (i != params.size()) {
 			url += "&amp;"
 		}
 		i++;
+		if ( key == "LABEL" ) {
+			labelValue = "LABEL=${value}"
+		}
 	}
 	echo "Rerun in Grinder: ${url}"
-	currentBuild.description += "<br><a href=${url}>Rerun in Grinder</a> TARGET can be changed to run only the failed test target"
+	currentBuild.description += "<br><a href=${url}>Rerun in Grinder</a> TARGET to run only the failed test targets"
+	def sameMachineUrl = url.replace(labelValue,"LABEL=${NODE_NAME}")
+	echo "Rerun in Grinder on the same machine: ${sameMachineUrl}"
```

Comment:
could drop the 'the' to read to save on a few characters if you want:
"Rerun in Grinder on same machine"